File: java/Euler48.java
class Euler48 {

	private static long power(int b, int e) {
		long res = 1;
		for (int i=0; i<e && res != 0; i++) {
			// consider only last ten digits
			// also, avoids overflow
			res = (res * b) % 10000000000L;
		}
		return res;
	}
	
	public static void main(String[] args) {
		long res = 0;
		for (int i=1; i<=1000; i++) {
			res += power(i, i);
		}
		System.out.println(res % 10000000000L);
	}
}

File: cpp/euler48.cpp
#include<iostream>
using namespace std;

long long power(int b, int e) {
    long long res = 1;
    for (int i=0; i<e && res!=0; i++) {
        res = res * b % 10000000000LL;
    }
    return res;
}

int main() {
    long long result = 0;
    for (int i=1; i<1001;i++)
        result += power(i, i);
    cout << result % 10000000000LL << endl;
}
